Doc/reference/ldap.rst

```diff
@@ -462,26 +462,25 @@ The module defines the following exceptions:
    are instead turned into exceptions, raised as soon an the error condition
    is detected.
 
-   The exceptions are accompanied by a dictionary possibly
-   containing an string value for the key :py:const:`desc`
-   (giving an English description of the error class)
-   and/or a string value for the key :py:const:`info`
-   (giving a string containing more information that the server may have sent).
-
-   A third possible field of this dictionary is :py:const:`matched` and
-   is set to a truncated form of the name provided or alias dereferenced
-   for the lowest entry (object or alias) that was matched.
-
-   The field :py:const:`msgid` is set in the dictionary where the
-   exception can be associated with an asynchronous request.
-   This can be used in asynchronous code where :py:meth:`result()` raises the
-   result of an operation as an exception. For example, this is the case for
-   :py:meth:`compare()`, always raises the boolean result as an exception
-   (:py:exc:`COMPARE_TRUE` or :py:exc:`COMPARE_FALSE`).
-
-   Most exceptions from protocol results also carry the :py:attr:`errnum`
-   attribute.
-
+   The exceptions are accompanied by a dictionary with additional information.
+   All fields are optional and more fields may be added in the future.
+   Currently, ``python-ldap`` may set the following fields:
+
+   * ``'desc'``: string giving a description of the error class, as provided
+     by OpenLDAP's ``ldap_err2string``.
+   * ``'info'``: string containing more information that the server may
+     have sent. The value is server-specific: for example, the OpenLDAP server
+     may send different info messages than Active Directory or 389-DS.
+   * ``'errno'``: numeric error code.
```

Comment:
If I understand the code correctly than ``LDAPraise_for_message`` sets **two** different numeric error codes

* ``'errno'``: numeric errno from system call, libc, or C function calls (see :mod:`errno`)
* ``'result'``: numeric LDAP protocol error number or API error number

``result`` is set ``errnum`` converted to Python int, ``desc`` is basically ``ldap_err2string(errnum)``. ``errno`` and ``errno`` are different things.